staging/src/k8s.io/apiserver/pkg/apis/apiserver/validation/validation.go

```diff
@@ -173,19 +174,40 @@ func validateClaimValidationRules(rules []api.ClaimValidationRule, fldPath *fiel
 	var allErrs field.ErrorList
 
 	seenClaims := sets.NewString()
+	seenExpressions := sets.NewString()
 	for i, rule := range rules {
 		fldPath := fldPath.Index(i)
 
-		if len(rule.Claim) == 0 {
-			allErrs = append(allErrs, field.Required(fldPath.Child("claim"), "claim name is required"))
+		if len(rule.Claim) > 0 {
```

Comment:
Consider using a `switch` statement with the most common error cases as the first things you check.